specs/GetFavicon.md

```diff
@@ -0,0 +1,146 @@
+# Background
+A Favicon is an asset which is a part of a  webpage, and typically displayed on each tab. Developers would
+like to have an API which allows them to retrieve the Favicon of a webpage, if it has one, as well as get an update whenever
+the favicon has changed.
+
+# Description
+We propose a new Webview2 event which would allow developers access the current Favicon of a page, 
+as well as be notified when the favicon changes.
+
+# Examples
+## Win32 C++ Registering a listener for favicon changes
+    ```cpp
+        CHECK_FAILURE(m_webViewStaging4->add_FaviconChanged(
+            Callback<ICoreWebView2StagingFaviconChangedEventHandler>(
+                [this](ICoreWebView2Staging4* sender, IUnknown* args) -> HRESULT {
+                    
+                    wil::unique_cotaskmem_string url;
+                    CHECK_FAILURE(sender->get_FaviconUri(&url));
+
+                    SaveFaviconUrl(url);
+                    
+                    const int buff_size = 100000;
+                    BYTE* buff = new BYTE[buff_size];
+                    IStream* pStream = SHCreateMemStream(buff, buff_size);
+
+                    sender->GetFaviconAsync(
+                        COREWEBVIEW2_FAVICON_IMAGE_FORMAT_PNG, 
+                        pStream,
+                        Callback<
+                            ICoreWebView2StagingGetFaviconCompletedHandler>(
+                            [&pStream, this](HRESULT code) -> HRESULT {
+                                     Gdiplus::Bitmap* pBitmap = new Gdiplus::Bitmap(pStream);
+                                     HICON icon;
+                                     pBitmap->GetHICON(&icon);
+                                     SendMessage(
+                                         m_appWindow->GetMainWindow(), WM_SETICON, ICON_SMALL,
+                                         (LPARAM)icon);
+                                return S_OK;
+                            })
+                            .Get());
+
+                    return S_OK;
+                })
+                .Get(),
+            &m_faviconChangedToken));  
+    ```
+## .NET / WinRT Registering a listener for favicon changes
+```c#
+webView.CoreWebView2.FaviconUriChanged += (CoreWebView2 sender, Object arg) =>
+{
+    string  value = sender.faviconUri;
+    SaveFaviconUrl(url);
+    System.IO.Stream stream = new System.IO.MemoryStream();
+
+    sender.GetFaviconAsync(
+        CoreWebView2FaviconImageFormat.Jpeg,
+        stream,
+        (HRESULT code) => {
+            SaveBytes(stream.ToArray());
```

Comment:
The .NET sample should match the C++ sample in spirit. Rather than saving the stream, we should be displaying the icon somehow